### PR TITLE
Add a GitHub Actions CI/CD workflow for sanity

### PR DESCRIPTION
**Pull request by webknjaz**
This is based on the current community collection template. I've only enabled `sanity` checks for now.

### CODE REVIEW

File: .github/workflows/ansible-test.yml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+          # OPTIONAL command to run before invoking `ansible-test integration`
+          # pre-test-cmd:
+          target-python-version: ${{ matrix.python }}
+          testing-type: integration
+          # OPTIONAL If your integration tests require code
+          # from other collections, install them like this
+          test-deps: ansible.netcommon
+
+
+  check:  # This job does nothing and is only used for the branch protection
```
Comment (webknjaz) [inline .github/workflows/ansible-test.yml:189]: @samdoran this can be added to branch protection

**Comment by webknjaz** [comment]
@samdoran maybe merge now, adjust later?

**Comment by samdoran** [comment]
Yeah, probably a good strategy.

**Comment by webknjaz** [comment]
Looks like there's new sanity checks updates that need to be handled. I'll send a PR.

**Comment by webknjaz** [comment]
Created https://github.com/samdoran/ansible-collection-macos/pull/10 to address that.

**Comment by webknjaz** [comment]
Looks like the milestone branch is also unstable now, dropping it: https://github.com/samdoran/ansible-collection-macos/pull/11

**Pull request merged**